### PR TITLE
Update MSBuildProject to give better error if TFM can't be read

### PR DESCRIPTION
**Pull request by mjrousos**
Also, adds a readiness check that confirms input projects have valid TFMs

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -279,3 +279,6 @@ dotnet_diagnostic.CA1711.severity = none
 # CA1716: Using a reserved keyword as the name of a namespace makes it harder for consumers in other languages to use the namespace
 dotnet_diagnostic.CA1716.severity = none
 
+# SA1200: Using directives must be placed within namespace
```
Comment (twsouthwick) [inline .editorconfig:282]: I just started seeing this. Thanks!

File: tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Checks/MultiTargetingCheckTests.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public async Task OnlyValidTFMsPassCheck(IProject[] projects, bool isReady)
+        {
+            // Arrange
+            var context = new Mock<IUpgradeContext>();
+            context.Setup(c => c.Projects).Returns(projects);
+
+            var readyCheck = new MultiTargetingCheck(NullLogger<MultiTargetingCheck>.Instance);
```
Comment (twsouthwick) [inline tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Checks/MultiTargetingCheckTests.cs:25]: I've been using `AutoMock` for this and it simplifies refactoings (ie if more services are injected into the check)

**Review by twsouthwick — APPROVED** [review]

File: src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs [hunk 3]
```diff
@@ -255,19 +255,29 @@ public TargetFrameworkMoniker TFM
...
-                        .Single(e => e.Name == "TargetFramework")
-                        .Value;
+                    var targetFrameworkProperties = ProjectRoot.Properties.Where(e => e.Name.Equals("TargetFramework", StringComparison.Ordinal));
+                    var propCount = targetFrameworkProperties.Count();
 
-                    return new TargetFrameworkMoniker(value);
+                    if (propCount != 1)
+                    {
+                        _logger.LogCritical("SDK projects being upgraded must have exactly one TargetFramework property. Found {TargetFrameworkCount} TargetFramework properties.", propCount);
+                        throw new UpgradeException($"SDK projects being upgraded must have exactly one TargetFramework property. Found {propCount} TargetFramework properties.");
```
Comment (KSchlobohm) [inline src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs:264]: Does this get shown to the user? Should it come from a resource file?

File: tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Checks/MultiTargetingCheckTests.cs (hunk 2 above)
Comment (mjrousos) [reply tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Checks/MultiTargetingCheckTests.cs:25]: This one's simple enough that I didn't really need AutoMock. I see what you mean about making future refactorings easy, though. I'll update it to use AutoMock.

File: src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs (hunk 3 above)
Comment (KSchlobohm) [reply src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs:264]: #285

**Pull request merged**